packages/eas-cli/src/build/runBuildAndSubmit.ts

```diff
@@ -203,17 +130,27 @@ export async function runBuildAndSubmitAsync(
   const buildCtxByPlatform: { [p in AppPlatform]?: BuildContext<Platform> } = {};
 
   for (const buildProfile of buildProfiles) {
+    const { exp, projectId } = await getDynamicProjectConfigAsync({
+      env: buildProfile.profile.env,
+    });
```

Comment:
I don't have anything specific against, but is there a reason you moved it here?

Reply:
https://github.com/expo/eas-cli/blob/b22a8abd96a40f5937f6a912f107dc1ad0bec3fb/packages/eas-cli/src/build/runBuildAndSubmit.ts#L276

The `resourceClass` was required to create build context as well as `getDynamicProjectConfigAsync` (`resolveResourceClass` also uses data from `getDynamicProjectConfigAsync`). We used `getDynamicProjectConfigAsync` inside of the `createBuildContextAsync` function, so another option was to move resolving `resourceClass` to the `prepareAndStartBuild` function https://github.com/expo/eas-cli/blob/b22a8abd96a40f5937f6a912f107dc1ad0bec3fb/packages/eas-cli/src/build/runBuildAndSubmit.ts#L251 or `createBuildContextAsync` function itself. I decided the best place resolve the resource class is the top-level `runBuildAndSubmit` function for better visibility and that's why I extracted this logic above.

Do you have a better place for it in mind?

Reply:
What do you think about it now?